s3stream/src/main/java/com/automq/stream/s3/S3Storage.java

```diff
@@ -417,7 +419,7 @@ public CompletableFuture<Void> append(AppendContext context, StreamRecordBatch s
      * @return backoff status.
      */
     public boolean append0(AppendContext context, WalWriteRequest request, boolean fromBackoff) {
-        // TODO: storage status check, fast fail the request when storage closed.
+        checkStatus();
```

Comment:
It should be `#completeException(new IOException())`

Reply:
yes, that would be better

Reply:
> It should be `#completeException(new IOException())`
  yes, that would be better

Reply:
It should invoke WalWriteRequest.cf#completeException